pkg/utils/permission.go

```diff
@@ -0,0 +1,275 @@
+			},
+			RoleRef: rbacv1.RoleRef{
+				Kind: "ClusterRole",
+				Name: clusterRole.Name,
+			},
+			Subjects: []rbacv1.Subject{
+				{
+					Kind: rbacv1.UserKind,
+					Name: username,
+				},
+			},
+		})
+}
+
+func (p *permissionBuilder) BindClusterRoleToGroup(clusterRole *rbacv1.ClusterRole, userGroup string) RBACPermissionBuilder {
+	return p.WithStaticClusterRole(clusterRole).
+		WithStaticClusterRoleBinding(&rbacv1.ClusterRoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: clusterRole.Name, // The same name as the cluster role
+			},
+			RoleRef: rbacv1.RoleRef{
+				Kind: "ClusterRole",
+				Name: clusterRole.Name,
+			},
+			Subjects: []rbacv1.Subject{
+				{
+					Kind: rbacv1.GroupKind,
+					Name: userGroup,
+				},
+			},
+		})
+}
+
+func (p *permissionBuilder) BindRoleToUser(role *rbacv1.Role, username string) RBACPermissionBuilder {
+	return p.WithStaticRole(role).
+		WithStaticRoleBinding(&rbacv1.RoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: role.Name, // The same name as the cluster role
+			},
+			RoleRef: rbacv1.RoleRef{
+				Kind: "Role",
+				Name: role.Name,
+			},
+			Subjects: []rbacv1.Subject{
+				{
+					Kind: rbacv1.UserKind,
+					Name: username,
+				},
+			},
+		})
+}
+
+func (p *permissionBuilder) BindRoleToGroup(role *rbacv1.Role, userGroup string) RBACPermissionBuilder {
+	return p.WithStaticRole(role).
+		WithStaticRoleBinding(&rbacv1.RoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: role.Name, // The same name as the cluster role
+			},
+			RoleRef: rbacv1.RoleRef{
+				Kind: "Role",
+				Name: role.Name,
+			},
+			Subjects: []rbacv1.Subject{
+				{
+					Kind: rbacv1.GroupKind,
+					Name: userGroup,
+				},
+			},
+		})
+}
+
+func (p *permissionBuilder) WithStaticClusterRole(clusterRole *rbacv1.ClusterRole) RBACPermissionBuilder {
+	p.u.fns = append(p.u.fns, func(cluster *clusterv1.ManagedCluster, addon *addonapiv1alpha1.ManagedClusterAddOn) error {
+		ensureAddonOwnerReference(&clusterRole.ObjectMeta, addon)
+		exists := true
+		latest, err := p.kubeClient.RbacV1().ClusterRoles().Get(context.TODO(), clusterRole.Name, metav1.GetOptions{})
+		if err != nil {
+			if !apierrors.IsNotFound(err) {
+				return err
+			}
+			exists = false
+		}
+		if !exists {
+			_, err := p.kubeClient.RbacV1().ClusterRoles().Create(context.TODO(), clusterRole, metav1.CreateOptions{})
+			if err != nil {
+				return err
+			}
+		} else {
+			clusterRole.ResourceVersion = latest.ResourceVersion
+			_, err := p.kubeClient.RbacV1().ClusterRoles().Update(context.TODO(), clusterRole, metav1.UpdateOptions{})
+			if err != nil {
+				return err
+			}
+		}
+		return nil
+	})
+	return p
+}
+
+func (p *permissionBuilder) WithStaticClusterRoleBinding(binding *rbacv1.ClusterRoleBinding) RBACPermissionBuilder {
+	p.u.fns = append(p.u.fns, func(cluster *clusterv1.ManagedCluster, addon *addonapiv1alpha1.ManagedClusterAddOn) error {
+		ensureAddonOwnerReference(&binding.ObjectMeta, addon)
```

Comment:
We cannot add this since this is a clusterrole

Reply:
removed ownerref